### PR TITLE
Add notes for how to do patch releases

### PR DESCRIPTION
**Pull request by leighmcculloch**
### What
Add notes for how to do patch releases

### Why
Close https://github.com/stellar/actions/issues/23

### CODE REVIEW

File: README-rust-release.md [hunk 1]
```diff
@@ -18,9 +18,21 @@ The [Rust crates at @stellar][crates] are released using the following process:
...
+  - [3. Merge PR](#3-merge-pr) (for `main` branch releases)
   - [4. Create Release on GitHub](#4-create-release-on-github)
-  - [5. Monitor the Publish](#5-monitor-the-publish)
+  - [5. Close PR](#5-merge-pr) (for backport patch releases)
+  - [6. Monitor the Publish](#6-monitor-the-publish)
+
+Note that this release process leans into trunk-based releasing, where most
+releases occur directly from the `main` branch. If patches need to be made to
+past releases:
+1. Create a release by performing step 1 below.
```
Comment (graydon) [inline README-rust-release.md:31]: Step 1 below is bump-version, it does not create a release. I'm not sure how I'd follow these instructions.

**Review by graydon — CHANGES_REQUESTED** [review]
Trying to follow the instructions as written would have me very confused. Just going to the URL in step 1 does nothing -- it does not produce a release -- and I don't know what I'm supposed to be doing between steps 1 and 2 since we only "continue from step 2" in point 3..

File: README-rust-release.md (hunk 1 above)
Comment (leighmcculloch) [reply README-rust-release.md]: I reworded this point and the section. Can you take another look?

**Review by graydon — APPROVED** [review]

**Comment by graydon** [comment]
LGTM!

**Pull request merged**